### PR TITLE
Add message API rfc

### PR DESCRIPTION
**Pull request by themouette**

### CODE REVIEW

File: rfc/2015-06-10-message-api.markdown [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+A **Thread** is a group of **Messages** with additional pieces of information.
+Every **Thread** MUST provide the following information:
+
+* `total_count`: the number of messages
+* `unread_count`: the number of unread messages
+* `subject`: the thread subject. Empty string if none.
+* `contacts`: a list of links to contacts involved in the conversation
+* `importance_level`: the importance level associated to message
+* `privacy_index`: the privacy index associated to message
+* `messages`: a list of links to most recent messages. **<--- This is used for
```
Comment (gdchamal) [inline rfc/2015-06-10-message-api.markdown]: Thread.messages is not possible without important impact on performance if API give it this way.
There is a /threads/<thread_id>/messages route to use for that

Comment (themouette) [inline rfc/2015-06-10-message-api.markdown]: I cannot have a request per thread on the home page, typical browser limits to 6 parallel requests including HTML, Javascript and CSS files.

What alternative do we have?

Comment (esion) [inline rfc/2015-06-10-message-api.markdown]: May be calculate & give some partial infos : 
* nb unread & total messages
* date of last message
* if any file has been join
* ?

Comment (gdchamal) [inline rfc/2015-06-10-message-api.markdown]: at this time only complex data left to compute is unread_count. thread.text will contain last shortened message (ake the headline, was previously slug in my response).

Thread.messages exist as a list of message's uuid inside this thread (so total_count can be computed), also threads.attachments for a list of all attachments exist.

Have one or two more commits to fit that (so unread_count is the only information in the current RFC I can't get correctly and something to discuss around message.status that I modelize today as message.flags structure)

**Pull request merged**